Explorer/Assets/DCL/Infrastructure/ECS/Unity/Materials/Systems/ResetMaterialSystem.cs

```diff
@@ -31,16 +32,41 @@ public ResetMaterialSystem(World world, DestroyMaterial destroyMaterial, ISceneD
 
         protected override void Update(float t)
         {
-            ResetQuery(World);
-            World.Remove<PrimitiveMeshRendererComponent>(in Reset_QueryDescription);
+            ResetPrimitiveMeshQuery(World);
+            ResetGltfNodeQuery(World);
         }
 
         [Query]
         [None(typeof(PBMaterial))]
-        private void Reset(Entity entity, ref PrimitiveMeshRendererComponent meshRendererComponent, ref MaterialComponent materialComponent)
+        private void ResetPrimitiveMesh(Entity entity, ref PrimitiveMeshRendererComponent meshRendererComponent, ref MaterialComponent materialComponent)
         {
             meshRendererComponent.SetDefaultMaterial(sceneData.Geometry.CircumscribedPlanes, sceneData.Geometry.Height);
             ReleaseMaterial.Execute(entity, World, ref materialComponent, destroyMaterial);
+            World.Remove<MaterialComponent>(entity);
+        }
+
+        [Query]
+        private void ResetGltfNode(Entity entity, ref GltfNodeMaterialCleanupIntention cleanupIntention, ref MaterialComponent materialComponent)
+        {
+            var gltfContainer = World.TryGetRef<GltfContainerComponent>(cleanupIntention.ContainerEntity, out bool exists);
+            if (!exists) return;
+
+            // Reset all renderers to their original state
+            foreach (var renderer in cleanupIntention.Renderers)
+            {
+                if (gltfContainer.OriginalMaterials!.TryGetValue(renderer, out var originalMaterial))
+                    renderer.sharedMaterial = originalMaterial;
+            }
+
+            // Clean up the material component and remove the entity
+            ReleaseMaterial.Execute(entity, World, ref materialComponent, destroyMaterial);
+            World.Remove<PBMaterial>(entity);
```

Comment:
You don't need to waste time removing components if you are going to destroy the entity right away

Reply:
good point, refactored at https://github.com/decentraland/unity-explorer/pull/4754/commits/5eaac59d46118993c0195bff21d88c6dddb70e3d